examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1344 @@
+                            scratch_data.level_set_fe_values.get_quadrature(),
+                            scratch_data.level_set_fe_values.get_update_flags())
+      , non_matching_fe_values(fe_collection,
+                               quadrature_1D,
+                               region_update_flags,
+                               mesh_classifier,
+                               level_set_dof_handler,
+                               level_set)
+    {}
+
+    // The following objects are standard for computations on cells and their
+    // faces. In TraceFEM, we need these fitted quadratures in addition to
+    // NonMatching FEValues, which takes care of unfitted ones, mainly due to
+    // the stabilization terms.
+    FEValues<dim>          fe_values;
+    FEInterfaceValues<dim> fe_interface_values;
+
+    const NonMatching::RegionUpdateFlags    region_update_flags;
+    const Quadrature<1>                    &quadrature_1D;
+    const hp::FECollection<dim>            &fe_collection;
+    const NonMatching::MeshClassifier<dim> &mesh_classifier;
+    const DoFHandler<dim>                  &level_set_dof_handler;
+    const VectorType                       &level_set;
+
+    // In addition to FEValues an unfitted quadrature for the main
+    // solution, we need a cell quadrature
+    //  for the FE space of the level-set.
+    FEValues<dim>              level_set_fe_values;
+    NonMatching::FEValues<dim> non_matching_fe_values;
+  };
+
+
+  // @sect2{Copy objects for TraceFEM}
+  // This CopyData object is customized for TraceFEM.
+  //  In particular, the implementation of the normal-gradient volume
+  //  stabilization relies on it.
+
+  template <int dim>
+  struct TraceFEM_CopyData
+  {
+    FullMatrix<double>                   cell_matrix;
+    Vector<double>                       cell_rhs;
+    std::vector<types::global_dof_index> local_dof_indices;
+    unsigned int                         cell_index;
+    double                               cell_L2_error_sqr;
+    double                               cell_H1_error_sqr;
+    double                               cell_stab_error_sqr;
+
+    template <class Iterator>
+    void reinit(const Iterator &cell, const unsigned int dofs_per_cell)
+    {
+      cell_matrix.reinit(dofs_per_cell, dofs_per_cell);
+      cell_rhs.reinit(dofs_per_cell);
+      local_dof_indices.resize(dofs_per_cell);
+      cell->get_dof_indices(local_dof_indices);
+    }
+
+    template <class Iterator>
+    void reinit(const Iterator &cell)
+    {
+      cell_index          = cell->active_cell_index();
+      cell_L2_error_sqr   = 0;
+      cell_H1_error_sqr   = 0;
+      cell_stab_error_sqr = 0;
+    }
+  };
+
+  // @sect2{Normal-gradient stabilization form of TraceFEM}
+  // The following class corresponds to the stabilization form, its contribution
+  // to the global matrix and to the error.
+  template <int dim>
+  class NormalGradientVolumeStabilization
```

Comment:
I think this would be a good place to have an introduction to the kind of operations this class provides. For example, have a list of function names and a short discussion (with formulas) of what each function does, etc.